source/fundamentals/crud/read-operations/retrieve.txt

```diff
@@ -0,0 +1,242 @@
+
+.. tabs::
+
+   .. tab:: Asynchronous
+      :tabid: find-list-async
+
+      .. code-block:: csharp
+
+         var restaurants = await _restaurantsCollection.Find(filter).ToList();
+
+   .. tab:: Synchronous
+      :tabid: find-list-sync
+
+      .. code-block:: csharp
+
+         var restaurants = _restaurantsCollection.Find(filter).ToListAsync();
+
+To see a full example of using ``Find()`` to find multiple documents, see
+:ref:`csharp-retrieve-additional-information`.
+
+Find All Documents
+~~~~~~~~~~~~~~~~~~
+
+To find all documents in a collection, pass an empty
+``BsonDocument`` as a query filter. Then append the ``ToList`` or ``ToListAsync``
+method to return the documents as a ``List`` object, or the ``ToCursor`` or ``ToCursorAsync``
+method to return the documents as a ``Cursor``.
+
+To see a fully runnable example of using ``Find()`` to find all documents, see 
+:ref:`csharp-retrieve-additional-information`.
+
+Modify Find Behavior
+~~~~~~~~~~~~~~~~~~~~
+
+You can modify the behavior of ``Find()`` by passing
+in ``FindOptions``. If you don't specify any options, the driver uses the 
+default values for each option.
+
+You can configure the commonly used options with the following methods:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``BatchSize`` 
+     - | Gets or sets the number of documents to return per batch.
+
+   * - ``Collation`` 
+     - | Sets the collation options. 
+
+   * - ``Comment`` 
+     - | Sets the comment to the query.
+
+   * - ``Hint`` 
+     - | Sets the hint for which index to use.
+       | Default: ``0``
+
+   * - ``MaxTime`` 
+     - | Sets the maximum execution time on the server for this operation.
+
+To see a full list of available options, see 
+`FindOptions Properties <{+api-root+}/html/Properties_T_MongoDB_Driver_FindOptions.htm>`__.
+
+Example
+~~~~~~~
+
+The following example passes a query filter and ``FindOptions`` to the ``Find()``
+method which performs the following actions:
+
+- Finds all documents with "Pizza" in the ``cuisine`` field
+- Sets the ``BatchSize`` to ``3``
+- Stores the results in a ``Cursor`` object
+- Prints the number of documents currently held in the ``Cursor``
+
+.. io-code-block::
+   :copyable: true
+
+   .. input::
+      :language: csharp
+
+      var filter = Builders<Restaurant>.Filter.Eq("cuisine", "Pizza");
+
+      var findOptions = new FindOptions
+      {
+         BatchSize = 3
+      };
+
+      using (var cursor = _restaurantsCollection.Find(filter, findOptions).ToCursor())
+      {
+         cursor.MoveNext();
+         WriteLine($"Number of documents in cursor: {cursor.Current.Count()}");
+      }
+
+   .. output::
+      
+      Number of documents in cursor: 3
+
+.. tip::
+
+   Create a ``Cursor`` with a `using statement <https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-statement>`__ to
+   ensure the proper disposal of resources.
```

Comment:
Issue: I don't know what is meant by "ensure the proper disposal of resources". Could this be clarified? What resources are disposed? What method does a "using statement" let you avoid calling?

Reply:
I tried to clarify this a little bit better. Let me know if this gets the point across. If not and if it seems too confusing or starts to get too cumbersome to try and describe here then I'm also open to the idea of removing it entirely. It may be more fitting in a future different page

Reply:
I like this tip and hope you keep it. I like your update as well! Only potential tweak I'd make is to provide a link to the API documentation for `Dispose` at the end of the admonition.